gitlab/resource_gitlab_repository_file.go

```diff
@@ -0,0 +1,153 @@
+package gitlab
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabRepositoryFile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabRepositoryFileCreate,
+		Read:   resourceGitlabRepositoryFileRead,
+		Update: resourceGitlabRepositoryFileUpdate,
+		Delete: resourceGitlabRepositoryFileDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"file": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"branch": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"author_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"author_email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"commit_message": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceGitlabRepositoryFileCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	file := d.Get("file").(string)
+
+	options := &gitlab.CreateFileOptions{
+		Branch:        gitlab.String(d.Get("branch").(string)),
+		AuthorEmail:   gitlab.String(d.Get("author_email").(string)),
+		AuthorName:    gitlab.String(d.Get("author_name").(string)),
+		Content:       gitlab.String(d.Get("content").(string)),
+		CommitMessage: gitlab.String(d.Get("commit_message").(string)),
+		Encoding:      gitlab.String("base64"),
```

Comment:
Based on my reading of the [API docs](https://docs.gitlab.com/ee/api/repository_files.html#create-new-file-in-repository) the encoding defaults to text, meaning this resource has a different default than the API. Could you add an `encoding` attribute and default it to `text`?

Reply:
When I initially started working on this I did have the encoding as something that could be set, however it derped when you passed in `text`

Reply:
Yep I can confirm this, passing in "text" to `encoding` returns a 400. It also seems that all file content is base64 encoded even if you pass plaintext as `content` and emit `encoding` entirely.

Reply:
@armsnyder Can you offer any clarity here it seems there some misaligned expectations on what we should do as the request to default to `text` encoding seems to cause a 400 error as base64 seems to be the only valid encoding?

Reply:
@nicholasklick probably should have cc'd you in the above comment as well. Looks like use of anything other than base64 for encoding is causing issues?